packages/sysvars/src/last-restart-slot.ts

```diff
@@ -0,0 +1,43 @@
+import { assertAccountExists, decodeAccount, type FetchAccountConfig } from '@solana/accounts';
+import { type Codec, getStructCodec, getU64Codec } from '@solana/codecs';
+import type { GetAccountInfoApi } from '@solana/rpc-api';
+import type { Rpc } from '@solana/rpc-spec';
+import type { Slot } from '@solana/rpc-types';
+
+import { fetchEncodedSysvarAccount, SYSVAR_LAST_RESTART_SLOT_ADDRESS } from './sysvar';
+
+/**
+ * The `LastRestartSlot` sysvar.
+ *
+ * Information about the last restart slot (hard fork).
+ *
+ * The `LastRestartSlot` sysvar provides access to the last restart slot kept in the
+ * bank fork for the slot on the fork that executes the current transaction.
+ * In case there was no fork it returns `0`.
+ */
+export type SysvarLastRestartSlot = Readonly<{
+    lastRestartSlot: Slot;
+}>;
+
+export function getSysvarLastRestartSlotCodec(): Codec<SysvarLastRestartSlot> {
```

Comment:
nit, you could make these return types a little bit stricter since they all seem to be of fixed-size.

```ts
// Before.
export function getSysvarLastRestartSlotCodec(): Codec<SysvarLastRestartSlot>;

// After.
export function getSysvarLastRestartSlotCodec(): FixedSizeCodec<SysvarLastRestartSlot, 8>;
```

Also note that TypeScript should already be able to infer that information _but_ I would still be explicit about it so a) the types are easier to read for the end-user and b) there's a bit of a performance issue at the moment that I'll aim to fix soon (see https://github.com/solana-labs/solana-web3.js/issues/2295).

Reply:
All but two are fixed size, but I'll scope them all!

Reply:
Strange, where are you seeing this error?

Reply:
<img width="581" alt="Screenshot 2024-03-13 at 12 29 12 PM" src="https://github.com/solana-labs/solana-web3.js/assets/67205063/8e46cc26-6dc9-4d07-954a-9543735490c8">

Reply:
It works if I cast it, though, but I'm not sure if that's expected behavior.

Reply:
It seems like the decoder builder is resolving the fixed size to `40` based on the fields provided, but the type passed in is read as `number`, so it's trying to compare a literal `40` type to `number`.